src/redisearch_api.c

```diff
@@ -123,17 +131,21 @@ static Document* RS_CreateDocument(const void* docKey, size_t len, double score,
 
 static int RS_DropDocument(IndexSpec* sp, const void* docKey, size_t len) {
   RedisModuleString* docId = RedisModule_CreateString(NULL, docKey, len);
+  int rc = 0;
   t_docId id = DocTable_GetIdR(&sp->docs, docId);
   if (id == 0) {
-    RedisModule_FreeString(NULL, docId);
-    return 0;
-  }
-  int rc = DocTable_DeleteR(&sp->docs, docId);
-  if (rc) {
-    sp->stats.numDocuments--;
-    return 1;
+    rc = 0;
```

Comment:
rc is already zero, just do if(id != 0) without any else.

Reply:
I think I maybe intended for rc to be REDISMODULE_OK and REDISMODULE_ERR, like all the other APIs. Let me change this.